File: problem_solving_for_beginners/c_programming/palindrome_number.c
#include<stdio.h>
/**
 * palindrome_number - checks wheather a number is a palindrom or not
 * @num: number to check
 */

void palindrome_number(int num)
{
    int tmp, mod, ans = 0;

    tmp = num;
    while (num >= 1)
    {
        mod = num % 10;
        ans = ans * 10 + mod;
        num = num / 10;
    }
    
    if (tmp == ans)
        printf("Palindrome\n");
    else
        printf("Not Palindrome\n");
}

/**
 * main - Entry point
 *
 * Return: 0 always (success)
 */
int main(void)
{
    int num;
    printf("Enter a positive number: ");
    scanf("%d", &num);

    palindrome(num);
    return 0;
}

File: problem_solving_for_beginners/Number_Problems/python/palindrome_number.py
def palindrome_number(num):
    ans = 0
    tmp = num
    
    while num >= 1:
        mod = num % 10
        ans = ans * 10 + mod
        num = num // 10
    
    if tmp == ans:
        print("Palindrome")
    else:
        print("Not Palindrome")

num = int(input("Enter a positive number: "))
palindrome_number(num)
